src/gui/transferlistsortmodel.cpp

```diff
@@ -244,12 +255,6 @@ bool TransferListSortModel::lessThan_impl(const QModelIndex &left, const QModelI
 
         return invokeLessThanForColumn(TransferListModel::TR_QUEUE_POSITION);
     }
-
-    // Finally, sort by hash
-    const TransferListModel *model = qobject_cast<TransferListModel *>(sourceModel());
-    const QString hashL = model->torrentHandle(left)->hash();
-    const QString hashR = model->torrentHandle(right)->hash();
-    return hashL < hashR;
```

Comment:
What's the benefit in this change?

Reply:
Such that the `switch()` would contain all possible cases, this way the code looks nicer to me.

Reply:
At least, this looks suspicious (and unreliable) if the function does not have an unconditional `return` operator (i.e. outside of any `if` or `switch` statement). AFAIK, code analysers dislike such code too.

Reply:
Maybe you can think of this way: 
```c++
int func() {
    switch(anything) {
    case 1:
        return 1;
    case default:
        return 0;
    }
}
```
No matter what `anything` is, all unhanded case falls into the `default` case. And for the function, it is guaranteed to have a return value. Does this convince you?

Reply:
It is confusing for me to see that a `switch()` has a `default` case but also have more code under the `switch()` block. That makes the `default` case redundant IMO (in our case here).

Reply:
>Maybe you can think of this way

But tomorrow someone will add another case that will not contain `return`... so this code is poorly maintainable and unreliable.
>It is confusing for me to see that a switch() has a default case but also have more code under the switch() block. That makes the default case redundant IMO (in our case here).

IMO, it's better to remove `default ` case then have no "unconditional" `return`.

Reply:
>IMO, it's better to remove default case then have no "unconditional" return.

I can agree with it. Let's leave it for another time, this PR is invalidated :P